src/lib/findCommonError.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Common errors and how to detect them
+ * @type    {Array}
+ */
+const CommonErrors = [
+  {
+    message: /^Forbidden:?(.+)?/i,
+    status: 401,
+    level: 'warn',
+    defaultText: 'Please login first',
+  },
+  {
+    message: /^Permission\s?Denied:?(.+)?/i,
+    status: 403,
+    level: 'warn',
+    defaultText: 'You do not have permission to access this.',
+  },
+  {
+    message: /^Not\s?Found:?(.+)?/i,
+    status: 404,
+    level: 'warn',
+    defaultText: 'Unable to find resource',
+  },
+  {
+    message: /^Bad\s?Request:?(.+)?/i,
+    status: 400,
+    level: 'warn',
+    defaultText: 'Bad Request',
+  },
+  {
+    code: /^EBADCSRFTOKEN$/,
+    status: 400,
+    level: 'warn',
+    defaultText: 'Bad Token',
+  },
+];
+
+module.exports.CommonErrors = CommonErrors;
+
+/**
+ * Parse the error and see if its a common error. If it's common we'll probably
+ * change the response status
+ * @param    {Error}    err
+ * @param    {object?}   logger
+ * @return   {Object}
+ */
+module.exports.findCommonError = function findCommonError(err, logger = undefined) {
+  // Attempt to match some common errors so we can apply the right status
+  return CommonErrors.map(commonError => {
+    const keys = Object.keys(commonError);
+    for (let k = 0; k < keys.length; k += 1) {
```

Comment:
The fix is switching from `Object.keys(commonError).forEach(...` to a standard for loop so we can return properly for the parent map